pkg/operation/botanist/apiserver.go

```diff
@@ -0,0 +1,225 @@
+
+	var nodeNetwork *net.IPNet
+	if b.Shoot.GetNodeNetwork() != nil {
+		_, nodeNetwork, err = net.ParseCIDR(*b.Shoot.GetNodeNetwork())
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse Node CIDR %q", *b.Shoot.GetNodeNetwork())
+		}
+	}
+
+	return kubeapiserverdeployment.New(
+		b.Shoot.Info.Spec.Kubernetes.KubeAPIServer,
+		b.ManagedSeedAPIServer,
+		b.K8sSeedClient,
+		b.K8sGardenClient.Client(),
+		b.Shoot.KubernetesVersion,
+		b.Shoot.SeedNamespace,
+		b.Shoot.Info.Namespace,
+		b.Shoot.HibernationEnabled,
+		b.Shoot.KonnectivityTunnelEnabled,
+		etcdEncryptionEnabled,
+		gardencorev1beta1helper.ShootWantsBasicAuthentication(b.Shoot.Info),
+		hvpaEnabled,
+		gardenletfeatures.FeatureGate.Enabled(features.MountHostCADirectories),
+		b.Shoot.Info.DeletionTimestamp != nil,
+		b.Shoot.Networks.Services,
+		b.Shoot.Networks.Pods,
+		nodeNetwork,
+		b.Shoot.GetMinNodeCount(),
+		b.Shoot.GetMaxNodeCount(),
+		b.Shoot.Info.Annotations,
+		b.Shoot.Info.Spec.Maintenance.TimeWindow,
+		sniValues,
+		images,
+	), nil
+}
+
+// DeployKubeControllerManager deploys the Shoot Kubernetes API Server and resources the kube-apiserver deployment depends on.
+func (b *Botanist) DeployKubeAPIServer(ctx context.Context) error {
+	b.Shoot.Components.ControlPlane.KubeAPIServer.SetSecrets(kubeapiserverdeployment.Secrets{
+		CA:                           component.Secret{Name: v1beta1constants.SecretNameCACluster, Checksum: b.CheckSums[v1beta1constants.SecretNameCACluster]},
+		CAFrontProxy:                 component.Secret{Name: kubeapiserverdeployment.SecretNameCAFrontProxy, Checksum: b.CheckSums[kubeapiserverdeployment.SecretNameCAFrontProxy]},
+		TLSServer:                    component.Secret{Name: kubeapiserverdeployment.SecretNameTLSServer, Checksum: b.CheckSums[kubeapiserverdeployment.SecretNameTLSServer]},
+		KubeAggregator:               component.Secret{Name: kubeapiserverdeployment.SecretNameKubeAggregator, Checksum: b.CheckSums[kubeapiserverdeployment.SecretNameKubeAggregator]},
+		KubeAPIServerKubelet:         component.Secret{Name: kubeapiserverdeployment.SecretNameKubeAPIserverKubelet, Checksum: b.CheckSums[kubeapiserverdeployment.SecretNameKubeAPIserverKubelet]},
+		StaticToken:                  component.Secret{Name: kubeapiserverdeployment.StaticTokenSecretName, Checksum: b.CheckSums[kubeapiserverdeployment.StaticTokenSecretName]},
+		ServiceAccountKey:            component.Secret{Name: v1beta1constants.SecretNameServiceAccountKey, Checksum: b.CheckSums[v1beta1constants.SecretNameServiceAccountKey]},
+		EtcdCA:                       component.Secret{Name: v1beta1constants.SecretNameCAETCD, Checksum: b.CheckSums[v1beta1constants.SecretNameCAETCD]},
+		EtcdClientTLS:                component.Secret{Name: etcd.SecretNameClient, Checksum: b.CheckSums[etcd.SecretNameClient]},
+		BasicAuth:                    component.Secret{Name: kubeapiserverdeployment.BasicAuthSecretName, Checksum: b.CheckSums[kubeapiserverdeployment.BasicAuthSecretName]},
```

Comment:
Does this actually work when basic auth is disabled? Same for the other secrets below that are conditional (konnectivity, vpn, ...). Even if yes, wouldn't it be more clear to only add them when necessary?